packages/core/src/sync/service.ts

```diff
@@ -89,9 +90,15 @@ export const create = async ({
   const onRealtimeSyncEvent = (realtimeSyncEvent: RealtimeSyncEvent) => {
     switch (realtimeSyncEvent.type) {
       case "checkpoint": {
-        syncService.networkServices.find(
+        const networkService = syncService.networkServices.find(
           (ns) => ns.network.chainId === realtimeSyncEvent.chainId,
-        )!.realtime!.checkpoint = realtimeSyncEvent.checkpoint;
+        )!;
+
+        // "realtime" property may be undefined when `kill()` has been
+        // invoked but hasn't completed.
+        if (networkService.realtime === undefined) return;
```

Comment:
Would it be simpler to just use `?.` instead of `!.` above?

Reply:
There's actually two different things that can be undefined. `networkService` can be undefined depending on the result of `.find()`. However, it's an invariant that it will always be defined, which is why the `!` is used. 

The `realtime` property could be undefined if the case described in the comment occurs.